### PR TITLE
[merged] integrate hash_pass into commctl

### PR DESCRIPTION
**Pull request by jlebon**
- Add a "commctl create passhash"
- Allow passing the password via --password, --file, or using getpass
- Small fix in create_host
- Delete hash_pass_script.py
- Add tests (might have gone overboard there, was fun to experiment :))
- Delete test_hash_pass_script.py

Example usage:

```
$ commctl create passhash
Password:
<hash>

$ commctl create passhash --password mypass
<hash>

$ commctl create passhash --file mypassfile
<hash>
```

### CODE REVIEW

**Comment by jlebon** [comment]
Ahhh shoot, I hadn't run flake8.

**Comment by jlebon** [comment]
Ohh, it's not flake8 that failed. It can't find the config file for it. Shouldn't that be part of setUp?

**Comment by ashcrow** [comment]
@jlebon it actually looks like unittests failures.

`python setup.py nosetest` is the easy way to run them locally.

**Comment by jlebon** [comment]
Right, it's because I didn't mock the realpath to return the conf. Will fixup (though that kinda sounds weird, shouldn't we just pass it directly as -c/--config ?).

**Comment by ashcrow** [comment]
@jlebon either is fine I think. If you'd like to switch it over to passing in the path I'd accept it. If not, I'd say keep it consistent and use the mock method.

**Comment by jlebon** [comment]
Fixed! :arrow_up: 

Sorry, should've used a !fixup.

**Comment by ashcrow** [comment]
LGTM

@cgwalters-bot try

**Comment by ashcrow** [comment]
@cgwalters-bot r+

File: src/commctl/client_script.py [hunk 1]
```diff
@@ -255,7 +255,7 @@ def create_host(self, address, **kwargs):
         infile = kwargs['ssh-priv-key']
         b64_bytes = base64.b64encode(infile.read())
         data['ssh_priv_key'] = b64_bytes.decode()
-        if 'cluster' in kwargs:
+        if kwargs['cluster'] is not None:
```
Comment (mbarnes) [inline src/commctl/client_script.py:258]: Sorry, I just now looked at this.

The `'cluster'` argument is optional, so I think this is going to throw a `KeyError` if it's not there.

Maybe instead: `if kwargs.get('cluster') is not None:`

Comment (jlebon) [inline src/commctl/client_script.py:258]: Heh, that's what I thought as well. Turns out if it's not provided, it will be `None`.

Comment (ashcrow) [inline src/commctl/client_script.py:258]: I thought about that as well but won't it default to `None` when not set via `argparse`?

Comment (jlebon) [inline src/commctl/client_script.py:258]: I think this is the relevant documentation: https://docs.python.org/2/library/argparse.html#default

> All optional arguments and some positional arguments may be omitted at the command line. The default keyword argument of add_argument(), whose value defaults to None, specifies what value should be used if the command-line argument is not present.

Comment (mbarnes) [inline src/commctl/client_script.py:258]: Yeah, looks like you're right.  Didn't realize that about `argparse`.

**Pull request closed without merging**